### ISSUE TITLE
v4.2 Bad error reporting

### ISSUE THREAD

**Issue by RoYoMi** [MSEndpointMgr/ModernDriverManagement#156]
## Problem

For those of us using the [Configmgr Task Sequence Monitor](https://smsagent.blog/tools/configmgr-task-sequence-monitor/) we'd appreciate a better script termination reporting on failures.

For example this is what we see in when the script fails. The message reads `InnerTerminatingFailure`. The problem is that every single error will report this same message. This requires the user to then crack open the ApplyDriversPackage.log to see exactly what happened. 
![image](https://user-images.githubusercontent.com/8366040/163043083-bd1fcb8f-241c-4234-927d-0fa83427aa0c.png)

## Recommendation
We'd rather see a descriptive message like this ` - Validation failed with empty list of matched driver packages for 'ThinkPad X1 Yoga 2nd', script execution will be terminated` instead of the meaningless `InnerTerminatingFailure` .
With this type of message we could immediately start reviewing the root cause.
![image](https://user-images.githubusercontent.com/8366040/163042969-baab0a03-03a0-47e4-8a4c-e28253d422eb.png)

## Solution
In theory this could be accomplished by make the following changes every where you're calling `New-TerminatingErrorRecord`
![image](https://user-images.githubusercontent.com/8366040/163042334-6b47b700-2af7-436c-b69a-061eed71d884.png)

**Comment by TANielsen** [comment]
Yea, I hate seeing red in my smsts.log

**Comment by RoYoMi** [comment]
No doubt the red is bad, but I hate when the logs lack the necessary details to investigate a problem. There should be a happy compromise in how the script writes these logs.

**Comment by stephannn** [comment]
Good call. I added it to our version: https://github.com/MSEndpointMgr/ModernDriverManagement/pull/226
